nsqd/http.go

```diff
@@ -596,3 +600,41 @@ func (s *httpServer) printStats(stats []TopicStats, health string, startTime tim
 	}
 	return buf.Bytes()
 }
+
+type allowedOpts struct {
+	NSQLookupdTCPAddresses []string `json:"nsqlookupd_tcp_addresses"`
```

Comment:
Verbs:

* PATCH uses the add/remove properties
* PUT uses the replace properties
* GET returns, as it does now
* DELETE is unused, because that would be silly

I'm :-/ about the semantics because it's not _really_ REST and is more RPC in spirit. However, the REST way to do it would be to do

* `POST /config/nsqlookupdtcpaddresses/` to add an address
* `PUT /config/nsqlookupdtcpaddresses/` to replace the entire list
* `DELETE /config/nsqlookupdtcpaddresses/{address}` to remove an address

But that seems excessive, and as I'm thinking about it, I'm probably overcomplicating matters. Let's step back for a second:

Can we think of a single service discovery mechanism that would give us an atomic view of the world? If we know about a change, we're almost certainly going to want to update the state of the world in DNS and etcd, at the very least.

As long as nsqd is intelligent about diffing updates (e.g., if it's connected to A & B, and you update the addresses to be A, B, and C, it just connects to C. it doesn't disconnect from A & B) then I'm having trouble thinking of a scenario where I'd actually use this.

Sorry for the knee jerk here, I may be off base about whether this is actually useful.